DatadogRUM/Sources/RUMConfiguration.swift

```diff
@@ -294,13 +294,23 @@ extension RUM.Configuration.URLSessionTracking {
         /// - Parameters:
         ///   - hosts: The set of hosts to inject tracing headers. Note: Hosts must not include the "http(s)://" prefix.
         ///   - sampleRate: The sampling rate for tracing. Must be a value between `0.0` and `100.0`. Default: `20`.
-        case trace(hosts: Set<String>, sampleRate: Float = 20)
+        ///   - traceControlInjection: The strategy for injecting trace context into requests. Default: `.all`.
+        case trace(
+            hosts: Set<String>,
+            sampleRate: Float = 20,
+            traceControlInjection: TraceContextInjection = .all
```

Comment:
question/ I wonder if the `TraceContextInjection` type shouldn't be re-exported through `DatadogRUM` [like we do for other internal types](https://github.com/DataDog/dd-sdk-ios/blob/c044ef9c69369a44c772d395c838d8c6951c2ab3/DatadogRUM/Sources/RUMConfiguration.swift#L15). My feeling is we don't need it unless someone wants to define a property of this type in their code, which is not the case here. For extra check, we can call this API in [smoke tests](https://github.com/DataDog/dd-sdk-ios/blob/c044ef9c69369a44c772d395c838d8c6951c2ab3/dependency-manager-tests/spm/App/ViewController.swift#L20) to see if it works without imoprting `DatadogInternal`.

The same applies to [re-exporting types in `DatadogTrace`](https://github.com/DataDog/dd-sdk-ios/blob/c044ef9c69369a44c772d395c838d8c6951c2ab3/DatadogTrace/Sources/TraceConfiguration.swift#L11-L21).

Reply:
actually we should, else customer have to import internal. Good call.